x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/elastic/ElasticInferenceServiceAclResponseEntity.java

```diff
@@ -0,0 +1,151 @@
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.inference.InferenceResults;
+import org.elasticsearch.inference.InferenceServiceResults;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.inference.external.http.HttpResult;
+import org.elasticsearch.xpack.inference.external.request.Request;
+
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+
+public class ElasticInferenceServiceAclResponseEntity implements InferenceServiceResults {
+
+    public static final String NAME = "elastic_inference_service_acl_results";
+    public static final String COMPLETION = TaskType.COMPLETION.name().toLowerCase(Locale.ROOT);
+
+    @SuppressWarnings("unchecked")
+    public static ConstructingObjectParser<ElasticInferenceServiceAclResponseEntity, Void> PARSER = new ConstructingObjectParser<>(
+        ElasticInferenceServiceAclResponseEntity.class.getSimpleName(),
+        args -> new ElasticInferenceServiceAclResponseEntity((List<AllowedModel>) args[0])
+    );
+
+    static {
+        PARSER.declareObjectArray(constructorArg(), AllowedModel.ALLOWED_MODEL_PARSER::apply, new ParseField("allowed_models"));
+    }
+
+    public record AllowedModel(String modelName, EnumSet<TaskType> taskTypes) implements Writeable, ToXContentObject {
+
+        @SuppressWarnings("unchecked")
+        public static ConstructingObjectParser<AllowedModel, Void> ALLOWED_MODEL_PARSER = new ConstructingObjectParser<>(
+            AllowedModel.class.getSimpleName(),
+            args -> new AllowedModel((String) args[0], toTaskTypes((List<String>) args[1]))
+        );
+
+        static {
+            ALLOWED_MODEL_PARSER.declareString(constructorArg(), new ParseField("model_name"));
+            ALLOWED_MODEL_PARSER.declareStringArray(constructorArg(), new ParseField("task_types"));
+        }
+
+        private static EnumSet<TaskType> toTaskTypes(List<String> stringTaskTypes) {
+            var taskTypes = EnumSet.noneOf(TaskType.class);
+            for (String taskType : stringTaskTypes) {
+                taskTypes.add(TaskType.fromStringOrStatusException(taskType));
```

Comment:
If the task type is invalid we should ignore it, that could result in an empty `task_types` array. If that happens we should remove the model entry.